### ISSUE TITLE
Design in progress

### ISSUE THREAD

**Issue by abin-aot** [bcgov/foi-flow#5]
As suggested, This design is a minimalistic  design which can cater the immediate problem of processing complex excel files and PDF outputs related to it.

This design focused only on processing complex Excel files problem as a quick win solution, as time allowed and group's agreement. The big picture architecture and design follows.

![image.png](https://images.zenhubusercontent.com/607de5aafedfab08839427c4/dcdf911f-6b48-44b5-8b98-79b38ab8b87f)

**Comment by sumathi-thirumani-aot** [comment]
Review in progress.

**Comment by acoard-aot** [comment]
One thing I wanted to mention that is not self-evident from the design.  The goal here is to create microservices that  can both:

1. Provide immediate value at end of sprint (~2wks)
2. Can be used going forward with minimal re-writing.

The idea is to focus on a microservice that solves the immediate XLS-to-PDF issue.  We'd stand up a very barebones VueJS web frontend that basically just lets users upload files to be processed by the XLS-to-PDF service.  Basically if user uploads 4 XLS, they get 4 PDFs.  

This MVP version wouldn't have auth or save any files.  In the future, the web would grow to have auth and could become the entire VueJS/Python microservices for the backoffice web backend.  In the future we would also integrate directly with the LAN to avoid need of user file upload, but could still re-use the same XLS-to-PDF microservice.

**Comment by lmullane** [comment]
@acoard-aot, agree with this.

As mentioned, I'd like to explore whether the PDF document generator my previous team used will work for us.

**Comment by sumathi-thirumani-aot** [comment]
Please do the needful on listed items for clarity and discussion

Clarity requested items
1. What is the need for user to manually upload files (Is it futuristic scope ?)
Understanding: Ministries will create a folder on file drive with data . So; ideally there is no need for IAO Staff to upload file. It can automatically read from file system.
2. What is the arrow between SharedLan and webfront represents ?
3. Is the files going to be stored in FOI DB ? [Arrow between FOI DB & Backend api] or this represents only indexing information?
4. Where the outcome PDF will be stored upon ?

Thank you.

**Comment by abin-aot** [comment]
@sumathi-thirumani-aot 

Yes and Ideally, actually we need to capture the FOI Requests on a Database rather than monitoring it from emails, then rather than creating folder manually. The application should create the FOI Request with the identifier once IAO analyst approves the requests and ready for Call for records. Then as I suggested a "File watcher" will be there to automatically dedupe and process to a PDF output for redaction , harms assessment. 


But the above diagram , @acoard-aot explained on his comment earlier is just a "minimalistic design" to create a quick win for the IAO team to solve initial problems with problematic file types only and we will expand it to bigger architecture as we progress.

Arrow between Shared Lan & Web , represents that IAO analyst points the file path (excel, Calender) for processing and PDF generation of it.  

We can haven't decided whether we will store the PDF output or not. We have to think on it. Few ideas came into table -  @acoard-aot suggested to store the metadata and MD5 into DB. My personal suggestion was to save the output in a Cache/DB with index information, MD5, then archives after a period of idle time.
